pkg/daemons/control/tunnel.go

```diff
@@ -148,11 +150,21 @@ func (t *TunnelServer) onChangePod(podName string, pod *v1.Pod) (*v1.Pod, error)
 			for _, ip := range pod.Status.PodIPs {
 				if cidr, err := util.IPStringToIPNet(ip.IP); err == nil {
 					if pod.DeletionTimestamp != nil {
-						logrus.Debugf("Tunnel server egress proxy removing Node %s Pod IP %v", nodeName, cidr)
-						t.cidrs.Remove(*cidr)
+						nets, _ := t.cidrs.ContainingNetworks(cidr.IP)
+						if entry, ok := nets[0].(*tunnelEntry); ok {
```

Comment:
as per my prior review - do this as

`if nets, err := t.cidrs.ContainingNetworks(cidr.IP); err == nil && len(nets) > 0 {`